src/webgpu/web_platform/util.ts

```diff
@@ -13,113 +13,284 @@ declare global {
-
-  for (let i = 0; i < expectation.length; ++i) {
-    unorm8[i] = Math.round(expectation[i] * 255.0);
-  }
+const kBt709PixelValue = {
+  displayP3: {
+    red: { R: 0.917522190548533, G: 0.200307878409863, B: 0.13860420691359, A: 1.0 },
+    green: { R: 0.458329787229903, G: 0.985255869269063, B: 0.298345835592774, A: 1.0 },
+    blue: { R: -0.0000827756941124, G: -0.0000284371289164, B: 0.959637561951994, A: 1.0 },
+    yellow: { R: 1.00000470900876, G: 0.999994346502453, B: 0.330959103516173, A: 1.0 },
+  },
+  srgb: {
+    red: { R: 1.0, G: 0.0, B: 0.0, A: 1.0 },
+    green: { R: 0.0, G: 1.0, B: 0.0, A: 1.0 },
+    blue: { R: 0.0, G: 0.0, B: 1.0, A: 1.0 },
+    yellow: { R: 1.0, G: 1.0, B: 0.0, A: 1.0 },
+  },
+};
 
-  return new Uint8Array(unorm8.buffer);
+function videoTable<Table extends { readonly [K: string]: {} }>({
+  table,
+}: {
+  table: Table;
+}): {
+  readonly [F in keyof Table]: {
+    readonly [K in keyof Table[F]]: Table[F][K];
+  };
+} {
+  return Object.fromEntries(
+    Object.entries(table).map(([k, row]) => [k, { ...row }])
+    /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  ) as any;
 }
 
-// kVideoExpectations uses unorm8 results
-const kBt601Red = convertToUnorm8(kBt601PixelValue.red);
-const kBt601Green = convertToUnorm8(kBt601PixelValue.green);
-const kBt601Blue = convertToUnorm8(kBt601PixelValue.blue);
-const kBt601Yellow = convertToUnorm8(kBt601PixelValue.yellow);
-
-export const kVideoExpectations = [
-  {
-    videoName: 'four-colors-vp8-bt601.webm',
-    _redExpectation: kBt601Red,
-    _greenExpectation: kBt601Green,
-    _blueExpectation: kBt601Blue,
-    _yellowExpectation: kBt601Yellow,
-  },
-  {
-    videoName: 'four-colors-theora-bt601.ogv',
-    _redExpectation: kBt601Red,
-    _greenExpectation: kBt601Green,
-    _blueExpectation: kBt601Blue,
-    _yellowExpectation: kBt601Yellow,
-  },
-  {
-    videoName: 'four-colors-h264-bt601.mp4',
-    _redExpectation: kBt601Red,
-    _greenExpectation: kBt601Green,
-    _blueExpectation: kBt601Blue,
-    _yellowExpectation: kBt601Yellow,
-  },
-  {
-    videoName: 'four-colors-vp9-bt601.webm',
-    _redExpectation: kBt601Red,
-    _greenExpectation: kBt601Green,
-    _blueExpectation: kBt601Blue,
-    _yellowExpectation: kBt601Yellow,
-  },
-  {
-    videoName: 'four-colors-vp9-bt709.webm',
-    _redExpectation: new Uint8Array([255, 0, 0, 255]),
-    _greenExpectation: new Uint8Array([0, 255, 0, 255]),
-    _blueExpectation: new Uint8Array([0, 0, 255, 255]),
-    _yellowExpectation: new Uint8Array([255, 255, 0, 255]),
+export const kVideoInfo = videoTable({
+  table: {
+    'four-colors-vp8-bt601.webm': {
+      mimeType: 'video/webm; codecs=vp8',
+      presentColors: {
+        'display-p3': {
+          red: kBt601PixelValue.displayP3.red,
+          green: kBt601PixelValue.displayP3.green,
+          blue: kBt601PixelValue.displayP3.blue,
+          yellow: kBt601PixelValue.displayP3.yellow,
```

Comment:
Are `.red`/etc still used? Would it work to just use:
```js
{
  topLeftColor: kBt601PixelValue.displayP3.yellow,
  topRightColor: kBt601PixelValue.displayP3.red,
  bottomLeftColor: kBt601PixelValue.displayP3.blue,
  bottomRightColor: green: kBt601PixelValue.displayP3.green,
}
```
? Then we wouldn't need the special handling in `videoTable` to "reify" the getters.